DNN Platform/Library/Common/Globals.cs

```diff
@@ -4061,5 +3830,15 @@ private static bool FilenameMatchesExtensions(string filename, string strExtensi
 
             return result;
         }
+
+        /// <summary>
+        /// Resolves dependencies after the <see cref="Globals.DependencyProvider"/>
+        /// has been initialized. This should only be called once to resolve all
+        /// dependencies used by the <see cref="Globals"/> object.
+        /// </summary>
+        private static void OnDependencyProviderChanged(object sender, PropertyChangedEventArgs eventArguments)
+        {
+            applicationStatusInfo = DependencyProvider.GetRequiredService<IApplicationStatusInfo>();

```

Comment:
Creating a method to register dependencies once the provider has been configured is needed since the `DependencyProvider` uses our custom `LazyServiceProvider`. **It is critical that the `Global.cs` resolves only ONE instance of a dependency, as it can generate too many sql connections that causes DNN to stop functioning**. 

This finding has been documented - #3989